lazy_tensor_core/lazy_tensor_core/csrc/compiler/backend_impl_interface.h

```diff
@@ -70,22 +71,31 @@ class BackendImplInterface {
    * Device Configuration
    * */
 
-  virtual std::string GetDefaultDevice() const = 0;
+  // Set or get the default device type.
+  // For backends used with virtual c10:: Devices, this configures what real device type
+  // the backend should use, and matters if the backend supports more than one type of real
+  // device.
+  virtual torch_lazy_tensors::BackendDeviceType GetDefaultDeviceType() const = 0;
+  virtual void SetDefaultDeviceType(std::string device_type) const = 0;
 
-  virtual size_t GetNumDevices() const = 0;
+  // Query all available backend devices
+  virtual torch_lazy_tensors::BackendDevice GetBackendDevices() const = 0;
```

Comment:
Shouldn't this return a vector of devices?

Reply:
ah.. yea!